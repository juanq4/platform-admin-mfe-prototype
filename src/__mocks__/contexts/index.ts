// do not export "./Auth0Context.mock" it breaks any mocking of useAuth0
export * from "./AdminDataContext.mock";
export * from "./LDContextMock.mock";
export * from "./NimbusConfig.mock";
export * from "./UserContext.mock";
