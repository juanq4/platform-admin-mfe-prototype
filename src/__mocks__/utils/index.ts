export * from "./helpers";
export * from "./permissions";
export * from "./wrappers";
