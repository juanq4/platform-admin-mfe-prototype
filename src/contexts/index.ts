export * from "./admin";
export * from "./application";
export * from "./broadcast";
export * from "./user";
