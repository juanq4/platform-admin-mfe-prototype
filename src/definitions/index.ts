export * from "./entity.definition";
export * from "./organization.definition";
export * from "./role.definition";
export * from "./token.definition";
export * from "./typescript.definition";
export * from "./user.definition";
export * from "./team.definition";
