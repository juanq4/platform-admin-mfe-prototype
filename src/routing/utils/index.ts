export * from "./api";
export * from "./s3checks";
export * from "./versionManager";
