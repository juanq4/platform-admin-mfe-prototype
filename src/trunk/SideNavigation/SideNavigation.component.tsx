import { Navigation, NavigationTheme } from "@q4/nimbus-ui";
import { memo, useEffect } from "react";
import type { SideNavigationProps } from "./SideNavigation.definition";
import { ViewIdModel } from "./SideNavigation.definition";
import { Container } from "./SideNavigation.style";

const SideNavigation = memo(function SideNavigation(props: SideNavigationProps): JSX.Element {
  const { routes, showMobileNavigation, onToggleMobileNavigation, onCollapse } = props;
  //TODO Replace this when PLATFORM-353 is complete and bring back rendering logo
  //based on showMobileNavigation
  // const isDesktopView = useMediaQuery({ query: MediaQuery.large.min });

  useEffect(() => {
    const navbar = document.querySelector("nav.nui-navigation ");
    let observer: MutationObserver;
    if (navbar) {
      observer = new MutationObserver(() => {
        onCollapse?.(navbar.classList.contains("nui-navigation--collapsed"));
      });
      observer.observe(navbar, {
        attributes: true,
        attributeFilter: ["class"],
        attributeOldValue: true,
      });
    }
    return () => {
      if (observer) {
        observer.disconnect();
      }
    };
  });

  return (
    <Container>
      <Navigation
        id={ViewIdModel.navigation.id}
        logo=""
        responsive={true}
        disableActiveRoute
        transitioned={showMobileNavigation}
        toggleTransition={onToggleMobileNavigation}
        routes={routes}
        theme={NavigationTheme.White}
      />
    </Container>
  );
});

export { SideNavigation };
