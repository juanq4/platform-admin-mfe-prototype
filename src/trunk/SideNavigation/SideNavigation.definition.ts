import type { NavigationRoute } from "@q4/nimbus-ui";
import { IdModelBase, isNullOrWhiteSpace, NavigationIdModel } from "@q4/nimbus-ui";
import type { NavigationProps } from "../Root/Root.definition";

export class SideNavigationIdModel extends IdModelBase {
  navigation: NavigationIdModel;

  constructor(id: string) {
    super(id);

    if (isNullOrWhiteSpace(this.id)) {
      this.navigation = new NavigationIdModel("");

      return;
    }

    this.navigation = new NavigationIdModel(`${this.id}Navigation`);
  }
}

export const ViewId = "Q4PlatformShell";
export const ViewIdModel = new SideNavigationIdModel(ViewId);

export interface SideNavigationProps extends NavigationProps {
  routes: NavigationRoute[];
  onCollapse?: (isCollapsed: boolean) => void;
  showManagedOrgSelector: boolean;
}
