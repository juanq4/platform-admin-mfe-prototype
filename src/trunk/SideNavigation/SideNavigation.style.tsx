import styled from "@emotion/styled";
import { MediaQuery } from "@q4/nimbus-ui";

export const Container = styled.div`
  z-index: 1;

  .nui-navigation--collapsed {
    & .nui-navigation_route:not(.nui-navigation_route--active) {
      transition: color 0s !important;
    }

    .nui-navigation_label-wrapper {
      animation-name: slide-right !important;
      animation-duration: 0.3s;
      animation-timing-function: ease-in-out;
      animation-fill-mode: forwards;
    }

    @keyframes slide-right {
      from {
        left: 4px;
        padding: 0px 8px;
      }
      to {
        left: 18px;
        padding: 0px;
      }
    }
  }

  @media ${MediaQuery.medium.max} {
    .nui-navigation {
      position: fixed;
    }
  }
`;
