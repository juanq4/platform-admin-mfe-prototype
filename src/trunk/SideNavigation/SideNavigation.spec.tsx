import { MediaDeviceSize, mockResizeWindow, NavigationClassName } from "@q4/nimbus-ui";
import { BrowserRouter } from "react-router-dom";
import type { UserContextState } from "../../contexts";
import { UserContext } from "../../contexts";
import { Organization } from "../../definitions";
import { act, render, screen, waitFor } from "../../utils/testUtils";
import { SideNavigation } from "./SideNavigation.component";
import { ViewIdModel } from "./SideNavigation.definition";

describe("SideNavigation", () => {
  const orgs = [
    new Organization({ id: "newId1", name: "New Organization", identifiers: ["NOW.TSX"] }),
    new Organization({ id: "newId2", name: "New Royal Organization", identifiers: ["NRO.TSX"] }),
  ];
  const usrContext: Partial<UserContextState> = {
    linkedOrganizations: orgs,
    managedOrganization: orgs[0],
  };

  test("7391331: [Given] no props are provided [Expect] the component to render", () => {
    const { container } = render(
      <BrowserRouter>
        <SideNavigation
          routes={[]}
          showMobileNavigation={false}
          onToggleMobileNavigation={jest.fn()}
          showManagedOrgSelector={false}
        />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  test("7391332: [Given] the user updates the window width [Expect] the correct body width", () => {
    [MediaDeviceSize.medium.min, MediaDeviceSize.extraLarge.min, MediaDeviceSize.extraSmall.max].forEach((size) => {
      act(() => {
        mockResizeWindow(size);
      });

      expect(window.innerWidth).toBe(size);
    });
  });

  test("7391333: [Given] the user has a viewport <= 768px [Expect] side navigation to be be transitioned by default", () => {
    const { rerender } = render(
      <UserContext.Provider value={usrContext}>
        <BrowserRouter>
          <SideNavigation
            routes={[]}
            showMobileNavigation={false}
            onToggleMobileNavigation={jest.fn()}
            showManagedOrgSelector={false}
          />
        </BrowserRouter>
      </UserContext.Provider>,
    );

    act(() => {
      mockResizeWindow(MediaDeviceSize.extraSmall.max);
    });

    const navbar = screen.getByTestId(ViewIdModel.navigation.id);
    expect(navbar).toBeVisible();

    expect(navbar).toHaveClass(NavigationClassName.BaseWithHiddenModifier);

    rerender(
      <UserContext.Provider value={usrContext}>
        <BrowserRouter>
          <SideNavigation
            routes={[]}
            showMobileNavigation={true}
            onToggleMobileNavigation={jest.fn()}
            showManagedOrgSelector={false}
          />
        </BrowserRouter>
      </UserContext.Provider>,
    );

    expect(navbar).toHaveClass(NavigationClassName.BaseWithVisibleModifier);
  });

  test("7391334: [Given] navigation is collapsed [Expect] the change is handled", async () => {
    const handler = jest.fn();
    render(
      <BrowserRouter>
        <SideNavigation
          routes={[]}
          showMobileNavigation={false}
          onToggleMobileNavigation={jest.fn()}
          onCollapse={handler}
          showManagedOrgSelector={false}
        />
      </BrowserRouter>,
    );
    const navbar = screen.getByTestId(ViewIdModel.navigation.id);
    navbar.classList.add("nui-navigation--collapsed");
    await waitFor(() => expect(handler).toHaveBeenCalled());
  });
});
