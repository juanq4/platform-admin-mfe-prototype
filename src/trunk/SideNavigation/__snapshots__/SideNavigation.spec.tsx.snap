// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation 7391331: [Given] no props are provided [Expect] the component to render 1`] = `
.emotion-0 {
  z-index: 1;
}

.emotion-0 .nui-navigation--collapsed .nui-navigation_route:not(.nui-navigation_route--active) {
  -webkit-transition: color 0s!important;
  transition: color 0s!important;
}

.emotion-0 .nui-navigation--collapsed .nui-navigation_label-wrapper {
  -webkit-animation-name: slide-right!important;
  animation-name: slide-right!important;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

@media only screen and (max-width: 1023px) {
  .emotion-0 .nui-navigation {
    position: fixed;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <nav
      class="nui-navigation nui-navigation--white nui-navigation--responsive nui-navigation--hidden"
      id="Q4PlatformShellNavigation"
    >
      <ul
        class="nui-navigation_list"
      />
      <div
        class="nui-navigation_collapse-toggle"
      >
        <i
          class="ni-caret-left-4pt"
          id="Q4PlatformShellNavigationToggle"
        />
      </div>
    </nav>
  </div>
</div>
`;
